### PR TITLE
logging(task/processor): Add logging to identify task causing DB error

### PR DESCRIPTION
**Pull request by gagantrivedi**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/deployment/locally-api#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?

## Changes

Wrap run_tasks in try/except to avoid thread from dying 
Catch and log DatabaseError caused by tasks to identify it

## How did you test this code?

Running task processor manually on a task that raises an Integrity Error

### CODE REVIEW

File: api/task_processor/processor.py [hunk 1]
```diff
@@ -89,6 +90,13 @@ def _run_task(task: Task) -> typing.Optional[typing.Tuple[Task, TaskRun]]:
 
         task_run.finished_at = timezone.now()
         task.mark_success()
+    except DatabaseError as e:
+        logger.error(
+            "Database error while running task %s:  %s",
+            task.id,
+            e,
+            exc_info=True,
+        )
```
Comment (khvn26) [inline api/task_processor/processor.py:99]: I think we have to add `task.mark_falure()` here?

File: api/task_processor/threads.py [hunk 2]
```diff
@@ -24,7 +27,10 @@ def __init__(
     def run(self) -> None:
         while not self._stopped:
             self.last_checked_for_tasks = timezone.now()
-            run_tasks(self.queue_pop_size)
+            try:
+                run_tasks(self.queue_pop_size)
+            except Exception as e:
+                logger.exception(e)
             run_recurring_tasks(self.queue_pop_size)
```
Comment (khvn26) [inline api/task_processor/threads.py:34]: Do we want to wrap this line too?

Comment (gagantrivedi) [reply api/task_processor/threads.py:34]: No, that's fine, this whole try/catch will go away once we figure out the task that is causing the issue

File: api/task_processor/processor.py (hunk 1 above)
Comment (gagantrivedi) [reply api/task_processor/processor.py:99]: No, there is no point doing that since we don't have access to the database any more here

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**